WebApp/Controllers/HomeController.cs

```diff
@@ -107,74 +106,58 @@ public async Task<ActionResult> SendMail(string recipient, string subject, strin
 ";
             string message = String.Format(messagetemplate, subject, body, recipient);
 
-            
-            HttpClient client = new HttpClient();            
-            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "https://graph.microsoft.com/v1.0/me/microsoft.graph.sendMail");
-            
-            request.Content = new StringContent(message, Encoding.UTF8, "application/json");
-
-            // try to get token silently
-            string signedInUserID = ClaimsPrincipal.Current.FindFirst(ClaimTypes.NameIdentifier).Value;
-            TokenCache userTokenCache = new MSALSessionCache(signedInUserID, this.HttpContext).GetMsalCacheInstance();
-            ConfidentialClientApplication cca = new ConfidentialClientApplication(clientId, redirectUri, new ClientCredential(appKey), userTokenCache, null);
-            var accounts = await cca.GetAccountsAsync();
-            if (accounts.Any())
+            HttpClient client = new HttpClient();
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "https://graph.microsoft.com/v1.0/me/microsoft.graph.sendMail")
             {
-                string[] scopes = { "Mail.Send" };
-                try
-                {
-                    AuthenticationResult result = await cca.AcquireTokenSilentAsync(scopes, accounts.First());
+                Content = new StringContent(message, Encoding.UTF8, "application/json")
+            };
 
-                    request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", result.AccessToken);
-                    HttpResponseMessage response = await client.SendAsync(request);
 
-                    if (response.IsSuccessStatusCode)
-                    {
-                        ViewBag.AuthorizationRequest = null;
-                        return View("MailSent");
-                    }
-                }
-                catch (MsalUiRequiredException)
+            IConfidentialClientApplication app = MsalAppBuilder.BuildConfidentialClientApplication();
+            AuthenticationResult result = null;
+            var accounts = await app.GetAccountsAsync();
+            string[] scopes = { "Mail.Send" };
+
+            try
+            {
+                // try to get token silently
+                result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault()).ExecuteAsync().ConfigureAwait(false);
+            }
+            catch (Exception eee)
```

Comment:
Please add a comment to explain why we always expect a token to be in the cache here.

Reply:
@bgavrilMS could you please validate if the explanation that I wrote is good enough?